src/main/java/cloud/orbit/spring/OrbitActorsProperties.java

```diff
@@ -28,19 +28,33 @@
 
 package cloud.orbit.spring;
 
-import cloud.orbit.actors.Stage;
 import org.springframework.boot.context.properties.ConfigurationProperties;
 
+import cloud.orbit.actors.Stage;
+import cloud.orbit.actors.extensions.ActorExtension;
+
 import java.util.List;
 
 @ConfigurationProperties(prefix = "orbit.actors")
 public class OrbitActorsProperties {
+    private String basePackages;
     private String clusterName;
     private String nodeName;
     private Stage.StageMode stageMode;
+    private List<ActorExtension> extensions;
```

Comment:
> This class is meant to be instantiated from the Spring configuration runtime against the properties file. The Spring framework wouldn't know how to instantiate those ActorExtension's from the property file.

Not entirely true. The configuration properties class is used to bind external configuration. A .properties file is but one implementation of external configuration. There are other equivalent methods. For instance, if a application.yaml is defined Spring will auto-load the file using SnakeYAML, which supports objects of any type, including custom Java objects:
https://bitbucket.org/asomov/snakeyaml/wiki/Documentation#markdown-header-loading-yaml

As far as the "spring way" of doing it, you are correct that there should be some kind of discovery of classes that implement ActorExtension. However, this would require a broader change in orbit-spring and possibly orbit-runtime.

The reason I exposed extensions this way is to match support that orbit-hk2 provided. Stage.class has this field:
```java
@Config("orbit.actors.extensions")
private List<ActorExtension> extensions = new CopyOnWriteArrayList<>();
```
which allows external configuration of extensions the same way. For example:
```yaml
orbit.actors.extensions:
  - !!cloud.orbit.actors.extensions.json.InMemoryJSONStorageExtension {}
  - !!cloud.orbit.actors.extensions.metrics.dropwizard.MetricsExtension {
    metricsConfig: [
      !!cloud.orbit.actors.extensions.metrics.dropwizard.DatadogReporterConfig {
        statsdHost: localhost,
        statsdPort: 8125,
        mode: udp,
        period: 10,
        periodUnit: SECONDS
        }
    ]
  }
```
Are you arguing that this kind of support should be dropped? Including support for configuring extensions externally does not prevent configuring extensions as beans or components at runtime like so:
```java
@Component
public class MyExtension implements OrbitSpringConfigurationAddon {
    @Override
    public void configure(final Stage stage) {
        ...
```